### PR TITLE
Matrix Feedback

### PR DESCRIPTION
**Pull request by krille90**
### Purpose
As a preparation for Systems, e.g. Atmos, Matrix changes will now be registered and communicated to the System, e.g. opening / closing doors.

Also additional clean up was made. Removing dead code, Scene clean up, etc.

### Open Questions and Pre-Merge TODOs

- [x]  I read the contribution guidelines and the wiki.
- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  This PR does not include files without specific need to do so.
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  This PR has been tested in multiplayer

### CODE REVIEW

File: UnityProject/Assets/Scripts/Managers/MatrixManager.cs [hunk 1]
```diff
@@ -206,39 +271,56 @@ public struct MatrixInfo {
...
+
+		if (netId == NetworkInstanceId.Invalid)
+		{
+			Debug.LogWarning($"Invalid NetID for matrix {matrix.gameObject.name}!");
 		}
+
 		return netId;
 	}
-}
+}
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Managers/MatrixManager.cs:326]: ow, 80 extra lines

Comment (DooblyNoobly) [reply UnityProject/Assets/Scripts/Managers/MatrixManager.cs:326]: those are lines of progress

Comment (fomalsd) [reply UnityProject/Assets/Scripts/Managers/MatrixManager.cs:326]: nope, that's just formatting :(

**Comment by fomalsd** [comment]
This PR totally broke startup from Lobby: you spawn in the space and start dying, FoV doesn't work, tons of NREs @krille90

**Comment by krille90** [comment]
I will look into it

Edit: looks like it broke because I moved the MatrixManager under Managers and didnt apply it to the prefab. Fixed it with the next PR.

File: UnityProject/Assets/Scripts/Managers/MatrixManager.cs (hunk 1 above)
Comment (krille90) [reply UnityProject/Assets/Scripts/Managers/MatrixManager.cs:326]: well, yea, consisting formatting is nice, isn't it? :)

**Pull request merged**